addon/adapters/drf.js

```diff
@@ -58,45 +65,80 @@ export default DS.RESTAdapter.extend({
-   *
-   * @method ajaxError
-   * @param  {Object} jqXHR
-   * @return {DS.InvalidError} or {Object} jqXHR
-   */
-  ajaxError: function(jqXHR) {
-    var error = this._super(jqXHR);
+    Takes an ajax response, and returns the json payload or an error.
 
-    if (jqXHR && jqXHR.status === 400) {
+    By default this hook just returns the json payload passed to it.
+    You might want to override it in two cases:
 
-      var jsonErrors;
-      try {
-        jsonErrors = Ember.$.parseJSON(jqXHR.responseText);
-      } catch (SyntaxError) {
-        // This happens with some errors (e.g. 500).
-        return error;
-      }
+    1. Your API might return useful results in the response headers.
+    Response headers are passed in as the second argument.
+
+    2. Your API might return errors as successful responses with status code
+    200 and an Errors text or object. You can return a `DS.InvalidError` or a
+    `DS.AdapterError` (or a sub class) from this hook and it will automatically
+    reject the promise and put your record into the invalid or error state.
 
-      // The field errors need to be in an `errors` hash to ensure
-      // `extractErrors` / `normalizeErrors` functions get called
-      // on the serializer.
-      var convertedJsonErrors = {};
-      convertedJsonErrors['errors'] = jsonErrors;
-      return new DS.InvalidError(convertedJsonErrors);
+    Returning a `DS.InvalidError` from this method will cause the
+    record to transition into the `invalid` state and make the
+    `errors` object available on the record. When returning an
+    `DS.InvalidError` the store will attempt to normalize the error data
+    returned from the server using the serializer's `extractErrors`
+    method.
+
+    @method handleResponse
+    @param  {Number} status
+    @param  {Object} headers
+    @param  {Object} payload
+    @return {Object | DS.AdapterError} response
+  */
+  handleResponse: function(status, headers, payload) {
+    if (this.isSuccess(status, headers, payload)) {
+      return payload;
+    } else if (this.isInvalid(status, headers, payload)) {
+      return new DS.InvalidError(this._drfToJsonAPIValidationErrors(payload));
+    }
 
-    } else {
-      return error;
+    if (Object.getOwnPropertyNames(payload).length === 0) {
+      payload = '';
+    } else if (payload.detail) {
+      payload = payload.detail;
     }
+    let errors = this.normalizeErrorResponse(status, headers, payload);
+
+    if (ERROR_MESSAGES[status]) {
+      return new DS.AdapterError(errors, ERROR_MESSAGES[status]);
+    }
+    return new DS.AdapterError(errors);
   },
 
+  isInvalid: function(status) {
+    return status === 400;
+  },
+
+  _drfToJsonAPIValidationErrors: function(payload) {
+    let out = [];
+    for (let key in payload) {
+      if (payload.hasOwnProperty(key)) {
+        payload[key].forEach((error) => {
+          if (key === this.get('nonFieldErrorsKey')) {
+            out.push({
```

Comment:
I took another look at the spec, it says that the pointer must be used to identify an attribute: '/data/attributes/some_attr' or '/data' to point to a primary data object, so something that identify a resource (or list of), which looks like this:

```js
{
  "data": {
    "type": "articles",
    "id": "1"
  }
}
```

So I don't think our case maps to any of these